tests/test_agent.py

```diff
@@ -3276,98 +3290,808 @@ def deferred_tool(x: int) -> int:  # pragma: no cover
             ]
         )
 
-    def test_early_strategy_does_not_apply_to_tool_calls_without_final_tool(self):
-        """Test that 'early' strategy does not apply to tool calls without final tool."""
-        tool_called = []
-        agent = Agent(TestModel(), output_type=self.OutputType, end_strategy='early')
-
-        @agent.tool_plain
-        def regular_tool(x: int) -> int:
-            """A regular tool that should be called."""
-            tool_called.append('regular_tool')
-            return x
-
-        result = agent.run_sync('test early strategy with regular tool calls')
-        assert tool_called == ['regular_tool']
+    def test_early_strategy_does_not_call_additional_output_tools(self):
+        """Test that 'early' strategy does not execute additional output tool functions."""
+        output_tools_called: list[str] = []
 
-        tool_returns = [m for m in result.all_messages() if isinstance(m, ToolReturnPart)]
-        assert tool_returns == snapshot([])
+        def process_first(output: OutputType) -> OutputType:
+            """Process first output."""
+            output_tools_called.append('first')
+            return output
 
-    def test_multiple_final_result_are_validated_correctly(self):
-        """Tests that if multiple final results are returned, but one fails validation, the other is used."""
+        def process_second(output: OutputType) -> OutputType:  # pragma: no cover
+            """Process second output."""
+            output_tools_called.append('second')
+            return output
 
         def return_model(_: list[ModelMessage], info: AgentInfo) -> ModelResponse:
             assert info.output_tools is not None
             return ModelResponse(
                 parts=[
-                    ToolCallPart('final_result', {'bad_value': 'first'}, tool_call_id='first'),
-                    ToolCallPart('final_result', {'value': 'second'}, tool_call_id='second'),
+                    ToolCallPart('first_output', {'value': 'first'}),
```

Comment:
The diff is a bit hard to read because everything moves around; can you let me know if any of the old behavior changed meaningfully?

Reply:
No, not at all. And sorry for changing not-related things. The tests that already existed have the same behavior

@DouweM